### PR TITLE
tidb_query: Support Range in TiKV Coprocessor Streaming

### PR DESCRIPTION
**Pull request by mapleFU**
###  What have you changed?

To fix the bug in https://github.com/pingcap/tidb/issues/12131, tikv need to carry a `range` in TiKV coprocessor streaming response in error case.

###  What is the type of the changes?

Pick one of the following and delete the others:

- Bugfix (Fix https://github.com/pingcap/tidb/issues/12131)

###  How is the PR tested?
Please select the tests that you ran to verify your changes:

- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No.

###  Does this PR affect `tidb-ansible`?

No.

### CODE REVIEW

File: components/tidb_query/src/storage/range.rs [hunk 1]
```diff
@@ -58,6 +58,17 @@ impl std::fmt::Debug for IntervalRange {
     }
 }
 
+impl Into<KeyRange> for IntervalRange {
```
Comment (iosmanthus) [inline components/tidb_query/src/storage/range.rs]: Try to implement `From<IntervalRange>` instead? It will implement `Into<KeyRange>` automatically.

Comment (mapleFU) [reply components/tidb_query/src/storage/range.rs]: In our history code: https://github.com/tikv/tikv/blob/b7ddddba68b344127a0908034522c3d0306b3240/src/coprocessor/dag/mod.rs#L144

Here we just use `IntervalRange` to change our `KeyRange`. But now `KeyRange` may not involved `reverse` or not, so I don't know if it's proper to implement it.

**Pull request closed without merging**